packages/babel-helper-modules/src/index.js

```diff
@@ -231,6 +246,48 @@ const reexportGetter = template(`
 `);
 
 /**
+ * Build a statement declaring a variable that contains all of the exported
+ * variable names in an object so they can easily be referenced from an
+ * export * from statement to check for conflicts.
+ */
+function buildExportNameListDeclaration(
+  programPath: NodePath,
+  metadata: ModuleMetadata,
+) {
+  const exportedVars = Object.create(null);
+  for (const [, data] of metadata.local) {
+    for (const name of data.names) {
+      exportedVars[name] = true;
+    }
+  }
+
+  let hasRexport = false;
+  for (const [, data] of metadata.source) {
```

Comment:
`metadata.source.values()`

Reply:
I don't feel that strongly about it, but personally I tend toward this when iterating something with key/value pairs, but I'll change it.